core/scripts/cli/voting/displayStatus.js

```diff
@@ -0,0 +1,87 @@
+  const currentTime = await voting.getCurrentTime();
+  const account = await getDefaultAccount(web3);
+  const filteredRequests = await filterRequests(pendingRequests, account, roundId, roundPhase, voting);
+  const rewards = await getAvailableRewards(web3, voting, account);
+  style.spinnerReadingContracts.stop();
+
+  // If no reveals have taken place in the current vote phase, then
+  // show the global inflation and GAT percentages. Otherwise,
+  // show the round's inflation and GAT percentages.
+  const _inflationRate =
+    roundStats.snapshotId.toString() === "0" ? await voting.inflationRate() : roundStats.inflationRate.toString();
+  const _gatPercentage =
+    roundStats.snapshotId.toString() === "0" ? await voting.gatPercentage() : roundStats.gatPercentage.toString();
+  const inflationRate = parseFloat(web3.utils.fromWei(_inflationRate)) * 100;
+  const gatPercentage = parseFloat(web3.utils.fromWei(_gatPercentage)) * 100;
+
+  // Compute time until next phase and round
+  const { minutesInLastHour, hoursUntilNextPhase, hoursUntilNextRound } = votePhaseTime(currentTime, roundPhase);
+
+  console.group(`${style.success(`\n** Your voting status **`)}`);
+  console.log(`${style.success(`- Current round ID`)}: ${roundId.toString()}`);
+  console.log(
+    `${style.success(`- Current round phase`)}: ${
+      roundPhase.toString() === VotePhasesEnum.COMMIT ? "Commit" : "Reveal"
+    }`
+  );
+  console.log(`${style.success(`- Round Inflation`)}: ${inflationRate.toString()} %`);
+  console.log(`${style.success(`- Round GAT`)}: ${gatPercentage.toString()} %`);
+  console.log(`${style.success(`- Contract time`)}: ${style.formatSecondsToUtc(currentTime)}`);
+  console.log(
+    `${style.success(
+      `- Time until ${roundPhase.toString() === VotePhasesEnum.COMMIT ? "Reveal" : "Commit"}`
+    )} phase: ${hoursUntilNextPhase} hours, ${minutesInLastHour} minutes`
+  );
+  console.log(
+    `${style.success(`- Time until next voting round`)}: ${hoursUntilNextRound} hours, ${minutesInLastHour} minutes`
+  );
+
+  // Display pending requests in a table
+  console.log(
+    `${style.success(`- Pending ${roundPhase.toString() === VotePhasesEnum.COMMIT ? "Price" : "Reveal"} Requests`)}:`
+  );
+  if (filteredRequests.length > 0) {
+    const requestsTable = [];
+    filteredRequests.forEach(request => {
+      const identifierUtf8 = web3.utils.hexToUtf8(request.identifier);
+      const timestampUtc = style.formatSecondsToUtc(parseInt(request.time));
+      requestsTable.push({
+        Identifier: identifierUtf8,
+        Time: timestampUtc
+      });
+    });
+    console.table(requestsTable);
+  }
+
+  // Display rewards to be retrieved in a table
+  console.log(`${style.success(`- Voting Rewards Available`)}:`);
+  if (rewards.roundIds.length > 0) {
+    const rewardsTable = [];
+    Object.keys(rewards.resolvedVotesByRoundId).forEach(roundId => {
+      rewards.resolvedVotesByRoundId[roundId].forEach(_reward => {
```

Comment:
Not sure if this is cleaner, but I thought I might throw another option out there for how to do this loop:
```
const reducer = (accumulator, currentValue) => accumulator + currentValue;
const rewardsTable = Object.values(rewards.resolvedVotesByRoundId).reduce(reducer).map(reward => {
 return {
      "Round ID": _reward.roundId,
      Name: _reward.name,
      "Reward Tokens": web3.utils.fromWei(_reward.potentialRewards)
    };
});
```
^After writing that out, it looks a little nasty too. I feel like this one just has to be kinda complicated :/

Reply:
yea i think the map reduce is cleaner but is less clear

Reply:
Nonetheless, I think you can use `Object.values()` so you don't have to re-access the object in the as you do on this line: `rewards.resolvedVotesByRoundId[roundId]`

Reply:
this doesn't work because this is the shape of `rewards.resolvedVotesByRoundId`

```
 {
    '9152': [
      {
        price: '122200000000000000000',
        name: 'bezos hairs @ August 15th 2019, 7:07:49 am',
        identifier: '0x62657a6f73206861697273000000000000000000000000000000000000000000',
        time: '1565852869',
        roundId: '9152',
        potentialRewards: '50050000000000000000000'
      }
    ],
    '9156': [
      {
        price: '0',
        name: 'population @ September 23rd 2019, 8:40:51 pm',
        identifier: '0x706f70756c6174696f6e00000000000000000000000000000000000000000000',
        time: '1569271251',
        roundId: '9156',
        potentialRewards: '50050000000000000000000'
      }
    ]
  }
```

Reply:
This reducer works however

```js
const reducer = (accumulator, currentValue) => accumulator.concat(currentValue);
```